components/omega/test/ocn/EosTest.cpp

```diff
@@ -0,0 +1,540 @@
+#include "TimeStepper.h"
+#include "Tracers.h"
+#include "mpi.h"
+
+// added for debug
+#include "AuxiliaryState.h"
+#include "Field.h"
+#include "Halo.h"
+#include "HorzMesh.h"
+
+#include <gswteos-10.h>
+
+using namespace OMEGA;
+
+// struct TestSetup {
+//    Real Radius = 6371220;
+//
+//    KOKKOS_FUNCTION Real layerThickness(Real Lon, Real Lat) const {
+//       return (2 + std::cos(Lon) * std::pow(std::cos(Lat), 4));
+//    }
+//
+//    KOKKOS_FUNCTION Real tracer(Real Lon, Real Lat) const {
+//       return (2 - std::cos(Lon) * std::pow(std::cos(Lat), 4));
+//    }
+// };
+
+constexpr Geometry Geom   = Geometry::Spherical;
+constexpr int NVertLevels = 60;
+// published values (TEOS-10) to test against
+const Real DeltaRefReal = 0.0009776149797;
+const Real VolRefReal   = 0.0009732819628;
+double Sa               = 30.;
+double Ct               = 10.;
+double P                = 1000.;
+const Real RTol         = 1e-10;
+// expected value for Linear eos (and default parameters)
+const Real LinearExpValue = 0.0009784735812133072;
+
+//------------------------------------------------------------------------------
+// The initialization routine for Eos testing. It calls various
+// init routines, including the creation of the default decomposition.
+// Not used now
+I4 initEosTest(const std::string &mesh) {
+
+   I4 Err = 0;
+
+   // Initialize the Machine Environment class - this also creates
+   // the default MachEnv. Then retrieve the default environment and
+   // some needed data members.
+   MachEnv::init(MPI_COMM_WORLD);
+   MachEnv *DefEnv  = MachEnv::getDefault();
+   MPI_Comm DefComm = DefEnv->getComm();
+
+   initLogging(DefEnv);
+   LOG_INFO("… in initEosTest");
+
+   // Open config file
+   Config("Omega");
+   Err = Config::readAll("omega.yml");
+   if (Err != 0) {
+      LOG_ERROR("Eos: Error reading config file");
+      return Err;
+   }
+
+   int TimeStepperErr = TimeStepper::init1();
+   if (TimeStepperErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing default time stepper");
+   }
+
+   int IOErr = IO::init(DefComm);
+   if (IOErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing parallel IO");
+   }
+
+   int DecompErr = Decomp::init(mesh);
+   if (DecompErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing default decomposition");
+   }
+
+   int HaloErr = Halo::init();
+   if (HaloErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing default halo");
+   }
+
+   int MeshErr = HorzMesh::init();
+   if (MeshErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing default mesh");
+   }
+
+   int TracerErr = Tracers::init();
+   if (TracerErr != 0) {
+      Err++;
+      LOG_ERROR("TendenciesTest: error initializing tracer infrastructure");
+   }
+
+   const auto &Mesh = HorzMesh::getDefault();
+   std::shared_ptr<Dimension> VertDim =
+       Dimension::create("NVertLevels", NVertLevels);
```

Comment:
We can get rid of this

Reply:
Removing this line led to a (non-critical) error in the `test/omega.log`:
`[* error] [Dimension.cpp:180] Cannot check distribution of dimension NVertLevels: dimension does not exist or has not been defined`
so I added it back. Let me know if you have a different syntax or suggestion...